index/full_car.go

```diff
@@ -0,0 +1,57 @@
+package index
```

Comment:
As I implement the DAG store, I realise that the FullIndex abstraction is actually unnecessary at this stage. In fact, it complicates things because when we obtain indices from the index repo to join with the data to create a read-only blockstore, and we do this through the carv2 library, which of course expects a carv2 index.

Conclusion: The index repo can custody CARv2 indices directly.